Runtime/Scripts/MediaStream.cs

```diff
@@ -45,29 +44,33 @@ public void Dispose()
             }
             if(self != IntPtr.Zero && !WebRTC.Context.IsNull)
             {
-                var tracks = GetTracks();
-                foreach (var track in tracks)
-                {
-                    StopTrack(track);
-                }
-                switch (_streamType)
-                {
-                    case MediaStreamType.Video:
-                        WebRTC.Context.DeleteVideoStream(self);
-                        break;
-                    case MediaStreamType.Audio:
-                        WebRTC.Context.DeleteAudioStream(self);
-                        break;
-                }
+                WebRTC.Context.DeleteMediaStream(this);
                 self = IntPtr.Zero;
             }
             this.disposed = true;
             GC.SuppressFinalize(this);
         }
 
-        public void FinalizeEncoder()
+        public DelegateOnAddTrack OnAddTrack
         {
-            WebRTC.Context.FinalizeEncoder();
+            get => onAddTrack;
+            set
+            {
+                onAddTrack = value;
+                selfOnAddTrack = MediaStreamOnAddTrack;
+                WebRTC.Context.MediaStreamRegisterOnAddTrack(self, selfOnAddTrack);
+            }
+        }
+
+        public DelegateOnRemoveTrack OnRemoveTrack
+        {
+            get => onRemoveTrack;
+            set
+            {
+                onRemoveTrack = value;
+                selfOnRemoveTrack = MediaStreamOnRemoveTrack;
```

Comment:
do we need to assign the static method to `selfOnRemoveTrack` and `selfOnAddTrack` ?

Reply:
Ooooh, I understood I wasn't thinking about anything...
I fixed it!